### PR TITLE
Fix indexes, use DocIndexState for fields shown in search

### PR DESCRIPTION
**Pull request by ThetaDR**
# Contribution checklist

## Checklist

* [ ] - UI test added to added/changed functionality?
* [ ] - Screenshot is added to PR if applicable ?
* [x] - Does a local formatting is applied (rush format)
* [x] - Does a local svele-check is applied (rush svelte-check)
* [ ] - Does a local UI tests are executed [UI Testing](../tests/readme.md)
* [x] - Does the code work? Check whether function and logic are correct.
* [ ] - Does Changelog.md is updated with changes?
* [ ] - Does the translations are up to date?
* [ ] - Does it well tested?
* [ ] - Tested for Chrome.
* [ ] - Tested for Safari.
* [ ] - Go through the changed code looking for typos, TODOs, commented LOCs, debugging pieces of code, etc.
* [ ] - Rebase your branch onto master and upstream branch
* [ ] - Is there any redundant or duplicate code?
* [ ] - Are required links are linked to PR?
* [ ] - Does new code is well documented ?

## Related issues

A list of closed updated issues

## Contributor requirements

* [x] - Sign-off is provided. [DCO](https://github.com/apps/dco)
* [x] - GPG Signature is provided. [GPG](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits)

### CODE REVIEW

File: server/core/src/mapper.ts [hunk 1]
```diff
@@ -13,22 +13,28 @@ interface IndexedReader {
...
+  doc: DocIndexState,
+  otherDocs?: Record<string, DocIndexState | undefined>,
   refAttribute?: string
 ): IndexedReader {
   return {
     get: (attr: string) => {
       const realAttr = hierarchy.findAttribute(_class, attr)
       if (realAttr !== undefined) {
-        return doc[docKey(attr, { refAttribute, _class: realAttr.attributeOf })]
+        // const attributes = doc._class === core.class.DocIndexState ? doc.attributes : doc
```
Comment (haiodo) [inline server/core/src/mapper.ts]:
```suggestion
```

File: server/core/src/mapper.ts [hunk 2]
```diff
@@ -13,22 +13,28 @@ interface IndexedReader {
...
+        return doc.attributes[docKey(attr, { refAttribute, _class: realAttr.attributeOf })]
       }
       return undefined
     },
     getDoc: (attr: string) => {
       const realAttr = hierarchy.findAttribute(_class, attr)
       if (realAttr !== undefined) {
-        const refAtrr = realAttr.type as RefTo<Doc>
-        return createIndexedReader(refAtrr.to, hierarchy, doc, docKey(attr, { _class }))
+        // const anotherDoc = doc._class === core.class.DocIndexState ? otherDocs?.[attr] : doc
```
Comment (haiodo) [inline server/core/src/mapper.ts]:
```suggestion
```

File: server/core/src/mapper.ts [hunk 3]
```diff
@@ -66,21 +72,32 @@ function findSearchPresenter (hierarchy: Hierarchy, _class: Ref<Class<Doc>>): Se
...
+    spaceDoc: DocIndexState | undefined
+  }
+): Promise<void> {
+  const searchPresenter = findSearchPresenter(hierarchy, elasticDoc._class)
   if (searchPresenter === undefined) {
     return
   }
 
-  const reader = createIndexedReader(doc._class, hierarchy, doc)
+  const reader = createIndexedReader(elasticDoc._class, hierarchy, doc, {
```
Comment (haiodo) [inline server/core/src/mapper.ts:87]: As I remember we discussed to get rid of reateIndexedReader and use Elastic DocIndexState attributes?

Comment (ThetaDR) [reply server/core/src/mapper.ts:87]: Yes. I kept it, but now it uses DocIndexState from mongo. I didn't want to split get and getDoc into two functions and kept it in one

Comment (haiodo) [reply server/core/src/mapper.ts:87]: I do not see changes related to it.

**Review by haiodo — APPROVED** [review]

**Pull request merged**